src/analysis/retail/monk/mistweaver/modules/spells/JadeEmpowerment.tsx

```diff
@@ -1,78 +1,166 @@
-      this.onCast,
+      Events.BeginChannel.by(SELECTED_PLAYER).spell(SPELLS.CRACKLING_JADE_LIGHTNING),
+      this.onCastStart,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.CRACKLING_JADE_LIGHTNING),
+      this.onCastEnd,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell([SPELLS.AT_HEAL, SPELLS.AT_CRIT_HEAL]),
+      this.handleATHeal,
     );
   }
 
-  onCast(event: CastEvent) {
-    const perfs: QualitativePerformance[] = [
-      this.selectedCombatant.hasBuff(SPELLS.JADE_EMPOWERMENT_BUFF)
-        ? QualitativePerformance.Good
-        : QualitativePerformance.Fail,
-    ];
-    if (this.hasJFT) {
-      perfs.push(
-        this.selectedCombatant.hasBuff(SPELLS.JT_BUFF)
-          ? QualitativePerformance.Good
-          : QualitativePerformance.Fail,
+  addValuesToTooltip() {
+    const lastEntry = this.castEntries[this.castEntries.length - 1];
+    if (lastEntry) {
+      const previousTooltip = lastEntry.tooltip;
+      lastEntry.tooltip = (
+        <>
+          {previousTooltip}
+          <br />
+          <div>
+            <SpellLink spell={SPELLS.ANCIENT_TEACHINGS} /> healing from this cast:{' '}
+            <b>{formatNumber(this.currentCJLHeal)}</b>
+          </div>
+        </>
       );
     }
+  }
+
+  onCastStart(event: BeginChannelEvent) {
+    if (this.currentCJLHeal > 0) {
+      this.addValuesToTooltip();
+    }
+    this.currentCJLHeal = 0;
+    this.insideCJLChannel = true;
+
+    const jadeEmpowermentPerf = this.selectedCombatant.hasBuff(SPELLS.JADE_EMPOWERMENT_BUFF)
+      ? QualitativePerformance.Good
+      : QualitativePerformance.Fail;
+
+    let jadefirePerf: QualitativePerformance | null = null;
+    if (this.hasJFT) {
+      jadefirePerf = this.selectedCombatant.hasBuff(SPELLS.JT_BUFF)
+        ? QualitativePerformance.Good
+        : QualitativePerformance.Fail;
+    }
+
+    let secretInfusionPerf: QualitativePerformance | null = null;
+    let activeSecretInfusionStat: string | null = null;
+
+    if (this.hasSI) {
+      if (this.selectedCombatant.hasBuff(SPELLS.SECRET_INFUSION_CRIT_BUFF)) {
+        activeSecretInfusionStat = 'Crit';
+        secretInfusionPerf = QualitativePerformance.Perfect;
+      } else if (this.selectedCombatant.hasBuff(SPELLS.SECRET_INFUSION_VERS_BUFF)) {
+        activeSecretInfusionStat = 'Vers';
+        secretInfusionPerf = QualitativePerformance.Perfect;
+      } else if (this.selectedCombatant.hasBuff(SPELLS.SECRET_INFUSION_HASTE_BUFF)) {
+        activeSecretInfusionStat = 'Haste';
+        secretInfusionPerf = QualitativePerformance.Good;
+      } else if (this.selectedCombatant.hasBuff(SPELLS.SECRET_INFUSION_MASTERY_BUFF)) {
+        activeSecretInfusionStat = 'Mastery';
+        secretInfusionPerf = QualitativePerformance.Ok;
+      } else {
+        secretInfusionPerf = QualitativePerformance.Ok;
+      }
+    }
+
+    const perfs: QualitativePerformance[] = [jadeEmpowermentPerf];
```

Comment:
You can move this line to line 93 and then instead of having to do the `(!== null)` checks below you can just do `perfs.push(secretInfusionPerf` inside the if block (and same for `jadefirePerf`)